Microsoft.Toolkit.Uwp.UI.Controls/RemoteDevicePicker/RemoteDevicePicker.cs

```diff
@@ -0,0 +1,374 @@
+        {
+            if (_listDevices.SelectedItems.Count > 0)
+            {
+                _listDevices.SelectedItems?.Clear();
+            }
+        }
+
+        /// <summary>
+        /// Initiate Picker UI
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        public async Task<IEnumerable<RemoteSystem>> PickDeviceAsync()
+        {
+            await ShowAsync();
+            IEnumerable<RemoteSystem> devices = ReturnDevices();
+            return devices;
+        }
+
+        internal IEnumerable<RemoteSystem> ReturnDevices()
+        {
+            List<RemoteSystem> returnValue = new List<RemoteSystem>();
+            foreach (RemoteSystem item in _listDevices.SelectedItems)
+            {
+                returnValue.Add(item);
+            }
+
+            return returnValue.AsEnumerable();
+        }
+
+        /// <inheritdoc />
+        protected override void OnApplyTemplate()
+        {
+            UnhookEvents();
+
+            _listDevices = GetTemplateChild("PART_ListDevices") as ListView;
+            _listDeviceTypes = GetTemplateChild("PART_ListDeviceTypes") as ComboBox;
+            _progressRing = GetTemplateChild("PART_ProgressRing") as ProgressRing;
+            _commandSpace = GetTemplateChild("CommandSpace") as Grid;
+            _deviceDiscovery = GetTemplateChild("DiscoveryType") as ComboBox;
+            _deviceStatus = GetTemplateChild("StatusType") as ComboBox;
+            _authorizationType = GetTemplateChild("AuthorizationType") as ComboBox;
+            _advancedFiltersGrid = GetTemplateChild("AdvancedFiltersGrid") as Grid;
+
+            var deviceList = typeof(RemoteSystemKinds).GetProperties().Select(a => a.Name).ToList();
+            deviceList.Add("All");
+            deviceList.Add("Unknown");
+
+            if (_listDeviceTypes != null)
+            {
+                _listDeviceTypes.ItemsSource = deviceList.OrderBy(a => a.ToString());
+                _listDeviceTypes.SelectionChanged += ListDeviceTypes_SelectionChanged;
+            }
+
+            if (_listDevices != null)
+            {
+                _listDevices.SelectionChanged += ListDevices_SelectionChanged;
+                _listDevices.ContainerContentChanging += ListDevices_ContainerContentChanging;
+
+                _listDevices.SelectionMode = SelectionMode == RemoteDeviceSelectionMode.Single ? ListViewSelectionMode.Single : ListViewSelectionMode.Multiple;
+                _listDevices.IsMultiSelectCheckBoxEnabled = SelectionMode == RemoteDeviceSelectionMode.Multiple ? true : false;
+                if (_listDevices.SelectionMode != ListViewSelectionMode.Single)
+                {
+                    _listDevices.DoubleTapped += ListDevices_DoubleTapped;
+                    if (_commandSpace != null)
+                    {
+                        _commandSpace.Visibility = Visibility.Visible;
```

Comment:
In single selection mode, I have no way to cancel when this is invisible now.

I'm not sure if `Esc` is supposed to work, but it may be an issue with focus?  I can't seem to keyboard focus on the dialog in the sample app to select an item with the keyboard?

Reply:
@michael-hawker I just tested this. `Esc works only if `CommandSpace` is enabled. That is the reason why in single mode it does not work. It works in multiple mode. I think the best way would be to show buttons by default and hide just press select and then access the system. Until we figure this out. I want to make sure stuff that is supposed to work a certain way does not break because of something that i am trying to make it work.